pkg/translator/jaeger/jaegerproto_to_traces.go

```diff
@@ -258,22 +258,24 @@ func jTagsToInternalAttributes(tags []model.KeyValue, dest pcommon.Map) {
 	}
 }
 
-func setInternalSpanStatus(attrs pcommon.Map, dest ptrace.Status) {
+func setInternalSpanStatus(attrs pcommon.Map, span ptrace.Span) {
+	dest := span.Status()
 	statusCode := ptrace.StatusCodeUnset
 	statusMessage := ""
 	statusExists := false
 
-	if errorVal, ok := attrs.Get(tracetranslator.TagError); ok {
+	if errorVal, ok := attrs.Get(tracetranslator.TagError); ok && errorVal.Type() == pcommon.ValueTypeBool {
+		statusCode = ptrace.StatusCodeOk
 		if errorVal.Bool() {
 			statusCode = ptrace.StatusCodeError
-			attrs.Remove(tracetranslator.TagError)
-			statusExists = true
+		}
+		attrs.Remove(tracetranslator.TagError)
```

Comment:
We should _not_ remove an attribute that has been explicitly added before us. Perhaps this is a 200 but an error occurred (the application might not be quite following REST principles).

The spec is a bit ambiguous here, as it states that it "must be left unset", but in this case, it was already set by something else (perhaps manually by developers? or using the attributes processor?).

Reply:
@jpkrohling So far we have been interchangeably treating jaeger `error` tag and OTel `StatusCode` in jaeger receiver/exporter with the exception that it worked only with error=true in the receiver due to the bug fixed in this PR. We set `error` jaeger tag back from the OTel `StatusCode` in the exporter https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7e904d87e53c5051b1eed1b2bd5b96788b16a645/pkg/translator/jaeger/traces_to_jaegerproto.go#L192. If we leave the `error` tag as an additional OTel attribute, I believe it'll result into duplicated jaeger `error` tag in the exporter. The correctness tests will also likely fail.

I'd suggest to keep the existing behavior and remove the tag but submit a separate issue where we discuss if it needs to be removed. PR to that issue would need to be applied to both receiver and exporter. WDYT?

Reply:
the changes got reverted.